manticore/core/memory.py

```diff
@@ -771,13 +772,54 @@ def read(self, addr, size):
 
         return result
 
+
+    def start_write_trace(self):
```

Comment:
imo naming these in a way that's explicit about a stack being involved would make this, and client code more readable. Maybe `push_record_writes`/`pop_record_writes`